VMEncryption/main/patch/redhatPatching.py

```diff
@@ -109,7 +101,16 @@ def install_extras(self):
             packages.remove('util-linux')
 
         if self.command_executor.Execute("rpm -q " + " ".join(packages)):
-            self.command_executor.Execute("yum install -y " + " ".join(packages))
+            ret_val = self.command_executor.Execute("yum install -y " + " ".join(packages))
+            if ret_val != CommonVariables.process_success and raise_on_failure:
+                msg = """
+                Failed to install all required dependencies.
+                Please check the internet connection, firewall settings
+                and package manager setup on the VM and try again.
+                Alternatively, install the following dependencies manually and then try again:
+                {}"""
+                msg = msg.format(str(packages))
+                raise Exception(msg)
```

Comment:
If we intend to fail early here, it may be desirable to simply log the failure here and exit with the CommonVariables.missing_dependency exit code so QOS telemetry can pick it up and avoid the risk of upstream caller of this function not catching or reporting the exception properly as a failure.

Reply:
good point, I'll make it so that the exit code gets sent out correctly.